parlai/scripts/generate_model_card.py

```diff
@@ -0,0 +1,1643 @@
+    'validation and logging info': ('valid', []),
+    'dictionary info/pre-processing': ('dict', []),
+    'other dataset-related info': (
+        None,
+        [
+            'fix_contractions',
+            'truncate',
+            'text_truncate',
+            'label_truncate',
+            'use_test_set',
+            'split_lines',
+            'balance_data',
+            'task',
+            'evaltask',
+        ],
+    ),
+    'more batch and learning rate info': (lambda x: 'batch' in x or 'lr' in x, []),
+    'training info': (
+        None,
+        [
+            'numthreads',
+            'shuffle',
+            'numworkers',
+            'metrics',
+            'gpu',
+            'data_parallel',
+            'optimizer',
+            'gradient_clip',
+            'adam_eps',
+            'nesterov',
+            'nus',
+            'betas',
+            'warmup_updates',
+            'warmup_rate',
+            'update_freq',
+            'fp16',
+            'max_train_time',
+            'no_cuda',
+        ],
+    ),
+    'pytorch info': ('pytorch', []),
+}
+
+
+USER_SYM_SECTION = 'user_included^-^:'
+
+# using a list for easier insertion + needs to be ordered.
+section_list = [
+    'model_details',
+    'model_details:_quick_usage',
+    'model_details:_sample_input_and_output',
+    USER_SYM_SECTION + 'intended_use',
+    USER_SYM_SECTION + 'limitations',
+    USER_SYM_SECTION + 'privacy',
+    'datasets_used',
+    'evaluation',
+    'extra_analysis',
+    USER_SYM_SECTION + 'related_paper',
+    'hyperparameters',
+    'feedback',
+]
+
+# sections that have unique functions
+defined_sections = {
+    'model_details',
+    'quick_usage',
+    'sample_input_and_output',
+    'datasets_used',
+    'evaluation',
+    'extra_analysis',
+    'quantitative_analyses',
+    'safety_benchmark',
+    'feedback',
+    'hyperparameters',
+}
+
+# default messages for certain sections
+defaults = {
+    "intended_use": "This model is intended for the use of....\t",
+    "privacy": "This model has the following privacy concerns....\t",
+    "limitations": "This model has has these limitations: ...\t",
+}
+
+# special sections that either have...
+special_section = {
+    # different than expected headings
+    'related_paper': "Related Paper(s)",
+    'evaluation': 'Evaluation Results',
+    # don't include titles
+    'extra_analysis': None,
+}
+
+# types of functions; I get the strings mixed ups easily
+CLASSIFIER = 'classifier'
+GENERATOR = 'generator'
+RETRIEVER = 'retriever'
+RANKER = 'ranker'
+
+# different modes
+M_gen = 'gen'
+M_edit = 'editing'
+M_final = 'final'
+
+# dictionary of all models with their path as the key
+all_models = {model['path']: model for model in model_list}
+# dictionary of all tasks with their task field as the key
+all_tasks = {task['task']: task for task in task_list}
+
+# set containing internal tasks where we can just remove "internal:" and it will be okay
+internal_remove_tasks = {
```

Comment:
I'd suggest we move the `internal_remove_tasks`  and `remap_task` to internal `igmc`. This is confusing to all public ParlAI users (as they have no idea what it means?)